### PR TITLE
Improve Legacy wins breakdowns migration and adviser matching.

### PR DESCRIPTION
**Pull request by elcct**
### Description of change

This accounts for an edge case where there are orphaned breakdowns in legacy dataset.

It also improves matching of advisers by replacing legacy domain names.

### Checklist

* [x] Has this branch been rebased on top of the current `main` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/main/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: datahub/export_win/test/test_legacy_migration.py [hunk 1]
```diff
@@ -735,3 +743,16 @@ def test_legacy_migration(mock_legacy_wins_pages):
     assert win_6.customer_response.other_marketing_source == ''
     assert win_6.line_manager == line_manager
     assert win_6.is_deleted is True
+
+
+@pytest.mark.parametrize(
+    'email,expected',
+    (
+        ('test@trade.gov.uk', {'test@businessandtrade.gov.uk', 'test@trade.gov.uk'}),
+        ('test@mobile.ukti.gov.uk', {'test@mobile.trade.gov.uk', 'test@trade.gov.uk'}),
```
Comment (oliverjwroberts) [inline datahub/export_win/test/test_legacy_migration.py:752]: I believe this wouldn't be an expected response as only `ukti` will be replaced in this example with the current implementation of `_email_mapping`. Is this the desired behaviour?

**Review by oliverjwroberts — COMMENTED** [review]
PR improves matching of adviser, consolidating the filtering/functionality to do so, as well as handles an edge case in the legacy dataset.

Looks good to me; just one comment regarding the failing test. Happy to approve once double checked.

**Review by p3dr0migue1 — APPROVED** [review]

File: datahub/export_win/test/test_legacy_migration.py (hunk 1 above)
Comment (elcct) [reply datahub/export_win/test/test_legacy_migration.py]: No, that was wrong. I updated the test now. Thank you!

**Review by oliverjwroberts — APPROVED** [review]
